foyer/smarts_graph.py

```diff
@@ -215,7 +216,7 @@ def __init__(self, G1, G2, node_match, element, typemap):
         if element not in [None, '*']:
             self.valid_nodes = [n for n, atom in 
                                 nx.get_node_attributes(G1, name='atom').items()
-                                if atom.element.symbol == element]
+                                if atom.element == md.element.get_by_symbol(element).atomic_number]
```

Comment:
This might already be taken care of since it is a parmed structure, but this could run into the same issues as before with improperly capitalized elements. 
```suggestion
                                if atom.element == md.element.get_by_symbol(element.capitalize()).atomic_number]
```